bench/tx-generator/src/Cardano/TxGenerator/Setup/NixService.hs

```diff
@@ -42,31 +53,76 @@ data NixServiceOptions = NixServiceOptions {
   , _nix_init_cooldown    :: Double
   , _nix_era              :: AnyCardanoEra
   , _nix_plutus           :: Maybe TxGenPlutusParams
+  , _nix_keepalive        :: Maybe Integer
   , _nix_nodeConfigFile       :: Maybe FilePath
   , _nix_cardanoTracerSocket  :: Maybe FilePath
   , _nix_sigKey               :: SigningKeyFile In
   , _nix_localNodeSocketPath  :: String
-  , _nix_targetNodes          :: NonEmpty NodeIPv4Address
+  , _nix_targetNodes          :: NonEmpty NodeDescription
   } deriving (Show, Eq)
 
 deriving instance Generic NixServiceOptions
 
+-- only works on JSON Object types
+data NodeDescription =
+  NodeDescription {
+      -- NodeIPAddress would be agnostic to IPv4 vs. IPv6 and likely
+      -- a small investment here.
+      ndAddr    :: NodeIPv4Address
+    , ndName    :: String
+    } deriving (Eq, Show, Generic)
+
+-- { "alias": "foo", "addr": ..., "port": ... }
```

Comment:
This is still a relic from the design phase :) Please change or remove, as it's now outdated.

Reply:
I also folded this back into the commit touching these lines in a rebase.